homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,250 @@
+"""Plugwise Climate component for HomeAssistant."""
+
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_OFF,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    TEMP_CELSIUS,
+)
+from homeassistant.exceptions import PlatformNotReady
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+_LOGGER = logging.getLogger(__name__)
+
+# Configuration directives
+CONF_MIN_TEMP = "min_temp"
+CONF_MAX_TEMP = "max_temp"
+
+# Default directives
+DEFAULT_NAME = "Plugwise Thermostat"
+DEFAULT_USERNAME = "smile"
+DEFAULT_TIMEOUT = 10
+DEFAULT_PORT = 80
+DEFAULT_ICON = "mdi:thermometer"
+DEFAULT_MIN_TEMP = 4
+DEFAULT_MAX_TEMP = 30
+
+# HVAC modes
+ATTR_HVAC_MODES = [HVAC_MODE_AUTO, HVAC_MODE_OFF]
+
+# Read platform configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_MIN_TEMP, default=DEFAULT_MIN_TEMP): cv.positive_int,
+        vol.Optional(CONF_MAX_TEMP, default=DEFAULT_MAX_TEMP): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
```

Comment:
Please rename `add_devices` to `add_entities`.

Reply:
Took this up together with the API init for the device to be handled